### PR TITLE
Remove some usages of slave from public fields.

### PR DESCRIPTION
**Pull request by jeffret-b**
I'm not sure if we want to remove the term in these instances. It depends on what we consider the API and how much we prioritize removing the term "slave" vs. maintaining any possible library use.

Essentially the API at this point are the command-line parameters (the `@Option` tags) and the object fields are internal (though they have the access modifier `public`). Any integration that steps beyond the CLI parameters into the fields is deep and rare.

I'm inclined to think it's better to make this change and continue the removal of the disfavored term. How concerned are others about maintaining the existing API at this level?

### CODE REVIEW

File: src/main/java/hudson/remoting/Engine.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public Thread newThread(final Runnable r) {
     @CheckForNull
     private URL hudsonUrl;
     private final String secretKey;
-    public final String slaveName;
+    public final String agentName;
```
Comment (jvz) [inline src/main/java/hudson/remoting/Engine.java]: If you're renaming it, you could make it private.

**Comment by oleg-nenashev** [comment]
FTR it could be tracked as a part of https://issues.jenkins-ci.org/browse/JENKINS-42816

**Comment by jeffret-b** [comment]
Thanks for reviewing, Oleg! I'd like to get things like this cleaned up where we can. Like you, I don't find any usages. It really doesn't make sense to support arbitrary integrations at this level.

I haven't yet decided whether I want to proceed with this one or continue to ignore the word usage.

**Comment by oleg-nenashev** [comment]
So let's also make the field private

**Comment by jeffret-b** [comment]
> So let's also make the field private

Agreed on that point.

File: src/main/java/hudson/remoting/Engine.java (hunk 1 above)
Comment (jeffret-b) [inline src/main/java/hudson/remoting/Engine.java]:
```suggestion
    private final String agentName;
```

**Pull request merged**